### PR TITLE
Use new scalastyle and run as part of compile task

### PR DESCRIPTION
**Pull request by taylorleese**

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -137,7 +138,10 @@ object BuildSettings {
         }
     },
     // scalaz-stream_2.10 is not on Maven Central, until that changes, this line needs to stay in
-    resolvers += "Scalaz Bintray Repo" at "http://dl.bintray.com/scalaz/releases",
+    resolvers += Resolver.bintrayRepo("scalaz", "releases"),
+    compileScalastyle := scalastyle.in(Compile).toTask("").value,
+    (compile in Compile) <<= (compile in Compile).dependsOn(compileScalastyle),
+    (scalastyleConfigUrl in Compile) := Option(url("https://raw.githubusercontent.com/paypal/cascade/develop/scalastyle-config.xml")),
```
Comment (arschles) [inline project/Build.scala]: can this use https://github.com/paypal/scala-style-guide/blob/develop/scalastyle-config.xml instead?

Comment (taylorleese) [inline project/Build.scala]: @arschles No, because that's the HTML page and not the raw file contents. It's still a reference to develop though.

Comment (arschles) [inline project/Build.scala]: what about https://raw.githubusercontent.com/paypal/scala-style-guide/develop/scalastyle-config.xml then

Comment (taylorleese) [inline project/Build.scala]: @arschles Ah yeah I should be using that repo now. I'll fix Cascade and Horizon to reference it.

**Pull request merged**